IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework.Tests/TestBuilderTests.cs

```diff
@@ -178,7 +178,54 @@ public void TestBlockWithMultipleExecuteMethodsThrows()
                 .AddDependencyInstance("Testing")
                 .AddTestBlock<ExampleTestBlockWithMultipleExecuteMethods>();
 
-            Assert.Throws<ArgumentException>(() => builder.ExecuteTestCase());
+            Assert.Throws<InvalidOperationException>(() => builder.ExecuteTestCase());
+        }
+
+        [Fact]
+        public void AddLoggerReturnsCorrectLogger()
+        {
+            TestBuilder builder = new TestBuilder();
+            builder
+                .AddLogger<ExampleLogger>()
+                .AddTestBlock<ExampleLoggerUsage>();
+
+            Assert.Throws<NotImplementedException>(() => builder.ExecuteTestCase());
+        }
+
+        [Fact]
+        public void AddFinallyBlockExecutesCorrectly()
+        {
+            TestBuilder builder = new TestBuilder();
+            builder
+                .AddTestBlock<ExampleTestBlockWithReturn>(false)
+                .AddFinallyBlock<ExampleFinallyBlock>()
+                .ExecuteTestCase();
+        }
+
+        // Actually... this probably shouldn't throw since it's a "finally" block meant to clean stuff up
+        // Figure out the right behavior and fix test before moving further with finally blocks
+        [Fact]
+        public void AddFinallyBlockExecutesCorrectlyWithNegativeCondition()
+        {
+            TestBuilder builder = new TestBuilder();
+            builder
+                .AddTestBlock<ExampleTestBlockWithReturn>(true)
+                .AddFinallyBlock<ExampleFinallyBlock>();
+
+            Assert.Throws<TrueException>(() => builder.ExecuteTestCase());
```

Comment:
`TrueException` is also a weird exception to be expecting here.

Reply:
I could probably tighten up the test / be more clear in the test name. Basically just need to make sure that the ExampleFinallyBlock throws when expected (_although_ after writing this test, I realized that's not the behavior we ultimately want, so I need to revisit.) The TrueException is what xUnit will throw if an Assert.True fails (which in this case is expected, but again, probably actually _not_ the desired behavior.)